### PR TITLE
Login modal

### PR DESCRIPTION
**Pull request by lionsdwarf**

### CODE REVIEW

**Comment by craigspaeth** [comment]
Mostly LGTM! The only comment that's blocking is the [`auth0` naming conflict](https://github.com/findOctave/cortex/pull/479#discussion_r249174919). Once that's resolved feel free to merge.

File: apps/marketing/pages/index.js [hunk 1]
```diff
@@ -78,8 +79,14 @@ class Home extends Component {
...
       }`
     )
+    const user = await getCurrentUser({ req })
+    return { ...data, user }
+  }
 
-    return { ...data }
+  async componentDidMount (props) {
+    this.setState({
+      user: this.props.user || (await getCurrentUser())
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply apps/marketing/pages/index.js]: Agreed!

File: lib/auth0/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/**
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply lib/auth0/index.js]: Doh! Thanks for catching. Will fixxy.

File: apps/portal/pages/profile.js [hunk 3]
```diff
@@ -1,27 +1,41 @@
...
     })
   }
 
   render () {
-    return this.state.user ? <Settings user={this.state.user} /> : <div />
+    return this.state.user ? (
+      <div>
+        <span
+          onClick={logout}
+          style={{
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply apps/portal/pages/profile.js]: ...slopped-together, to-be-replaced temp-code....

File: apps/marketing/components/Layout/index.js [hunk 4]
```diff
@@ -8,13 +8,20 @@ import Footer from '../Footer'
 
 const cx = classNames.bind(styles)
 
-const Layout = ({ page, metaData, footer, children }) => {
+const Layout = ({
+  page,
+  metaData,
+  footer,
+  children,
+  renderLoginModal,
```
[the remark this answers is not in the document]
Comment (craigspaeth) [reply apps/marketing/components/Layout/index.js]: Yeah 🤔 good point. Maybe the linter has trouble with destructuring (or maybe I'm overlooking something 🙃 )

File: lib/auth0/index.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  return new Auth0LockPasswordless(
+    process.env.AUTH0_CLIENT_ID,
+    process.env.AUTH0_DOMAIN,
+    {
+      prefill,
+      passwordlessMethod: 'code',
+      autofocus: true,
+      responseType: 'id_token',
+      auth: {
+        audience: 'https://www.findoctave.com/api',
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply lib/auth0/index.js]: I believe the `audience` param existed prior to this PR. Not sure why git disagrees....

File: lib/auth0/index.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+export const extractTokenFromHash = href => {
+  const params = {}
+  href.replace(/([^(?|#)=&]+)(=([^&]*))?/g, ($0, $1, $2, $3) => {
+    params[$1] = $3
+  })
+  return params.id_token
+}
+
+const isServerSide = req => req
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply lib/auth0/index.js]: Haha yea that's totally fair. I felt really dumb writing this function, but valued it's explicitness. You're right--this should definitely be relegated to a comment.

File: apps/portal/pages/profile.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (craigspaeth) [reply apps/portal/pages/profile.js]: Yep yep! Didn't intend that as criticism, just making note for us both to remember to clean up later.

File: lib/auth0/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (craigspaeth) [reply lib/auth0/index.js]: Yep yep! Ditt—didn't intend that as criticism, just making note for myself to remember to clean up later.

File: lib/auth0/index.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    JSCookie.set(AUTH_TOKEN, token)
+    jwt = jwtDecode(token)
+  }
+  const { options, query } = clientQuery(fields, jwt, token)
+  const gql = new GraphQLClient(process.env.APP_URL + '/api', options)
+  const { client } = await gql.request(query)
+  return { client, jwt }
+}
+
+export const renderLogin = ({ prefill }) => getLock(prefill).show()
```
[the remark this answers is not in the document]
Comment (lionsdwarf) [reply lib/auth0/index.js]: An artifact from old code. Will rm.

**Comment by lionsdwarf** [comment]
Oop, didn't realize there was a merge conflict here. Resolving now. @craigspaeth

File: apps/marketing/pages/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (craigspaeth) [reply apps/marketing/pages/index.js]: One thing I noticed later: You're not actually using `this.state.user` in this component. Only pulling it out of [props](https://github.com/findOctave/cortex/pull/479/files#diff-fa004c36f6a9a7258f36843e466c65a0R102)—maybe this was a copy + past issue?

**Pull request merged**